traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -767,6 +773,26 @@ JOIN type t ON s.type = t.id ` +
 	select max(last_updated) as t from last_deleted l where l.table_name='server') as res`
 }
 
+func getServerCount(tx *sqlx.Tx, query string, queryValues map[string]interface{}) (uint64, error) {
+	rowsAffected := 0
+	var serverCount uint64
+	countOrigRows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get servers count: %v", err)
+	}
+	defer countOrigRows.Close()
+	for countOrigRows.Next() {
+		if err = countOrigRows.Scan(&serverCount); err != nil {
+			return 0, fmt.Errorf("failed to read servers count: %v", err)
+		}
+		rowsAffected++
+	}
+	if rowsAffected != 1 {
+		return 0, fmt.Errorf("incorrect rows returned for server count, want: 1 got: %v", rowsAffected)
+	}
+	return serverCount, nil
```

Comment:
A scalar value is returned from getServerCount(). No need to defer or loop, this can compute the server count in SQL itself and return the result.

Reply:
Fixed to use a NamedStatement instead.